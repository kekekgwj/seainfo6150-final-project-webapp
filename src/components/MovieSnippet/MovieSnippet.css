.movie-snippet-container {
    display: block;
    text-align: center;
    margin: 0 25px 10px 0;
    width: 115px;
}

.cover-wrapper {
    position: relative;
    background-size: 100%;
    height: 160px;
    overflow: hidden;
}
.cover-wrapper::after {
    content: '';
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 20px;
    background-image: linear-gradient(rgba(0,0,0,0), rgba(0,0,0,0.5));
}

.cover-wrapper img{
    max-width: 100%;
    border-width: 0;
    vertical-align: middle;
}

.movie-snippet-info {
    margin: 5px 0 0;
    height: 44px;
    overflow: hidden;
    color: #37a;
    font-size: 13px;
}

.new {
    opacity: 1;
    position: relative;
    left: -5px;
    width: 16px;
    height: 16px;
    border-width: 0;
    vertical-align: middle;
}
.rating {
    position: relative;
    left: 5px;
    color: #e09015;
}