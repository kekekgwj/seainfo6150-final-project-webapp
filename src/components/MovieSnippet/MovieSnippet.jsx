import React from "react";
import "./MovieSnippet.css";
import newIcon from "../../assets/new-icon.png";
import { useHistory } from 'react-router-dom';

const MovieSnippet = (props) => {
    const { snippetData } = props;
    const {
        cover = "",
        isNew = false,
        name = "",
        rating = 0,
        id
    } = snippetData;

    const history = useHistory();
    const onClickCover = (e) => {
        if (e.target && e.target.id) {
            const id = e.target.id;
            history.replace(`/movie/${id}`)
        } 
    }
    return (
        <div className="movie-snippet-container">
            <div className="cover-wrapper"  onClick={(e) => onClickCover(e)}>
                <img src={cover} id={id} alt="movie-cover" ></img>
            </div>
            <p className="movie-snippet-info">
                {isNew && 
                    <span>
                        <img className="new" src={newIcon} alt="new-movie"></img>
                    </span> }
                <span>{name}</span>
                <span className="rating">{rating}</span>
            </p>
        </div>
    )
}

export default MovieSnippet;