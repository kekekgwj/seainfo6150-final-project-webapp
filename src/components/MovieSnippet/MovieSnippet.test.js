import React from "react";
import MovieSnippet from "./MovieSnippet";

const SnippetData = {
    cover: "http://localhost:5000/images/poster/p2636551330.jpeg",
    isNew: true,
    rating: 7,
    name : "nobody",
    id: "p2636551330",
}
describe("MovieSnippet tests", () => {
  it("renders correctly", () => {
    const { container } = render(<MovieSnippet snippetData={SnippetData} title="test title"/>);
    expect(container).toMatchSnapshot();
  });
});