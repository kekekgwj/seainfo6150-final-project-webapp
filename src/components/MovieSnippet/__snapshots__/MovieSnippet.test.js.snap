// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSnippet tests renders correctly 1`] = `
<div>
  <div
    class="movie-snippet-container"
  >
    <div
      class="cover-wrapper"
    >
      <img
        alt="movie-cover"
        id="p2636551330"
        src="http://localhost:5000/images/poster/p2636551330.jpeg"
      />
    </div>
    <p
      class="movie-snippet-info"
    >
      <span>
        <img
          alt="new-movie"
          class="new"
          src="new-icon.png"
        />
      </span>
      <span>
        nobody
      </span>
      <span
        class="rating"
      >
        7
      </span>
    </p>
  </div>
</div>
`;
